src/software/world/ball.h

```diff
@@ -38,9 +38,18 @@ class Ball final
 
     /**
      * Returns the current state of the ball
+     *
+     * @return TimestampedBallState
      */
     TimestampedBallState currentState() const;
 
+    /**
+     * Returns the ball model to predict future states
+     *
+     * @return ball model
+     */
+    const std::shared_ptr<BallModel> &getBallModel() const;
```

Comment:
See comment above, I don't think we should be publicly exposing the ball model.

Reply:
We're only exposing a pointer to a virtual ball model. The only thing callers will see is the `estimateFutureState` function. No implementation detail should be leaked